.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,73 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ main ]
+  schedule:
```

Comment:
Should we remove the `schedule` bit? Not sure why we need this? E.g. in [other CI](https://github.com/microsoft/coyote/blob/main/.github/workflows/build-test-linux.yml) we don't have this, I think `push` and `pull_request` is enough to run on a PR.

Reply:
The schedule is just run it again once a week, in case there are no checkins, this will pick up any new rules developed by codeql folks.

Reply:
Ohh, I see, that makes sense.